e2e/extensions-protocol.e2e.spec.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * This tests the extension protocol handler.
+ * An E2E test is required to have access to the web page context.
+ */
+
+import path from 'path';
+
+import { ElectronApplication, Page, test, expect } from '@playwright/test';
+
+import { NavPage } from './pages/nav-page';
+import {
+  createDefaultSettings, getResourceBinDir, retry, startRancherDesktop, teardown,
+} from './utils/TestUtils';
+
+import { ContainerEngine, Settings } from '@pkg/config/settings';
+import { spawnFile } from '@pkg/utils/childProcess';
+
+/** The top level source directory, assuming we're always from the tree */
+const srcDir = path.dirname(path.dirname(__filename));
+const rdctl = executable('rdctl');
+
+/**
+ * Get the given executable. Similar to @pkg/utils/resources, but does not use
+ * Electron.app (which doesn't work during the test).
+ */
+function executable(name: string) {
+  const exeName = name + (process.platform === 'win32' ? '.exe' : '');
+
+  return path.join(srcDir, 'resources', process.platform, 'bin', exeName);
+}
+
+test.describe.serial('Extensions protocol handler', () => {
+  let app: ElectronApplication;
+  let page: Page;
+  let isContainerd = false;
+
+  async function ctrctl(...args: string[]) {
```

Comment:
No please don't use `ctr` as an abbreviation for containerd. Maybe just spell it out in full...

Reply:
It is what we use in BATS to mean either `nerdctl` or `docker` depending on which engine you are running.

I'm happy to find a better name, but it has to be short, ideally less than 8 characters.

Reply:
ok. There was an abbreviation in Komodo for `controller` that I was never happy with, and don't remember now. I keep seeing `counter` for `ctr` but don't have a better solution, can learn to live with it.

Reply:
Yeah, I'm using `ctrctl` here because I'm following the naming in BATS (I did just write a BATS test right before working on this…)

I'm happy to change names, but I'd need to know what to change _to_, as well as doing so in BATS as well.

Reply:
> I keep seeing `counter` for `ctr`

FWIW, I was thinking `ConTaineR` when I was picking the name. `ctr` is also the name of an old `containerd` [client](https://man.archlinux.org/man/community/containerd/ctr.8.en), so `ctrctl` seemed somewhat related/appropriate 🤷